src/components/dataset/query/QueryView.jsx

```diff
@@ -77,8 +82,21 @@ export class QueryView extends React.PureComponent {
   }
 
   componentDidUpdate(prevProps, prevState) {
-    const { dataset, dispatch, filterStatement, joinStatement, orderBy } = this.props;
-    const { selected } = this.state;
+    const { dataset, dispatch, filterStatement, joinStatement, orderBy, profile } = this.props;
+    const { selected, canLink } = this.state;
+
+    if (
+      this.hasDataset() &&
+      dataset.defaultProfileId &&
+      canLink === false &&
+      Object.keys(prevProps.profile).length === 0
+    ) {
+      dispatch(getBillingProfileById(dataset.defaultProfileId));
+    }
+
+    if (profile.id && canLink === false) {
```

Comment:
Do you need the `canLink === false`?  I don't know if set state triggers an event if the value doesn't change.  Also, I think it's more idiomatic to use `!canLink`

Reply:
Right on!  I think that's what happens when you use a PureComponent (which ends up being a nice perf optimization)

Reply:
I don't think we do this today but if you check out this:
https://redux-saga.js.org/docs/advanced/Channels/

I think that this is what we want to do.  Basically, watch for the `GET_DATASET_BY_ID_SUCCESS` action and have that trigger fetching the profile